qiskit/transpiler/passes/optimization/remove_identity_equiv.py

```diff
@@ -0,0 +1,56 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Transpiler pass to drop gates with negligible effects."""
+
+from __future__ import annotations
+
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.transpiler.target import Target
+from qiskit.transpiler.basepasses import TransformationPass
+from qiskit._accelerate.remove_identity_equiv import remove_identity_equiv
+
+
+class RemoveIdentityEquivalent(TransformationPass):
+    """Remove gates with negligible effects.
+
+    Removes gates whose effect is close to an identity operation, up to the specified
+    tolerance.
+    """
```

Comment:
It would be nice to be concrete about when a gate is cutoff, e.g. something along the lines of 
```suggestion
    For a cutoff fidelity :math:`f`, this pass removes gates whose average 
    gate fidelity with respect to the identity is below :math:`f`. Concretely, 
    a gate :math:`G` is removed if :math:`\bar F < f` where 
    
    .. math::

        \bar{F} = \frac{1 + F_{\text{process}}{1 + d} 
        
        F_{\text{process}} = \frac{|\mathrm{Tr}(G)|^2}{d^2}
        
    where :math:`d = 2^n` is the dimension of the gate for :math:`n` qubits.
    """
```